toqito/helper/bell_notation_conversions.py

```diff
@@ -0,0 +1,748 @@
+            for x in range(ia):
+                for y in range(ib):
+                    for a in range(oa - 1):
+                        target_row_0based = _cg_row_index(a, x)
+                        for b in range(ob - 1):
+                            target_col_0based = _cg_col_index(b, y)
+                            cg_mat[target_row_0based, target_col_0based] = v_mat[a, b, x, y]
+
+    return cg_mat
+
+
+def fp_to_fc(v_mat: np.ndarray, behaviour: bool = False) -> np.ndarray:
+    r"""Convert a Bell functional or behaviour from Full Probability (FP) to Full Correlator (FC) notation.
+
+    Assumes binary outcomes (oa=2, ob=2). The FP tensor indices `a, b = 0, 1`
+    correspond to physical outcomes `+1, -1` respectively.
+
+    The Full Probability (FP) notation represents the full probability distribution
+    :math:`V(a, b, x, y) = P(\text{out}_A=a', \text{out}_B=b' | x, y)`, where
+    :math:`a=0 \rightarrow a'=+1`, :math:`a=1 \rightarrow a'=-1` (similarly for b),
+    stored as a 4D numpy array `V[a, b, x, y]`.
+    The Full Correlator (FC) notation is represented by a matrix (see :func:`fc_to_cg`).
+
+    This function converts from FP to FC notation.
+
+    Examples
+    ==========
+
+    Consider the CHSH inequality functional in FP notation:
+    (Here V represents coefficients, not probabilities)
+
+    >>> import numpy as np
+    >>> from toqito.helper import fp_to_fc, fc_to_fp
+    >>> chsh_fc = np.array([[0, 0, 0], [0, 0.25, -0.25], [0, -0.25, 0.25]])
+    >>> chsh_fp = fc_to_fp(chsh_fc)
+    >>> fp_to_fc(chsh_fp) # CHSH functional in FC
+    array([[ 0.  ,  0.  ,  0.  ],
+           [ 0.  ,  0.25, -0.25],
+           [ 0.  , -0.25,  0.25]])
+
+
+    Consider a behaviour (probability distribution) in FP notation (standard PR box):
+
+    >>> pr_box = np.zeros((2, 2, 2, 2))
+    >>> pr_box[0, 0, 0, 0] = 0.5 # p(0,0|0,0)
+    >>> pr_box[1, 1, 0, 0] = 0.5 # p(1,1|0,0)
+    >>> pr_box[0, 0, 0, 1] = 0.5 # p(0,0|0,1)
+    >>> pr_box[1, 1, 0, 1] = 0.5 # p(1,1|0,1)
+    >>> pr_box[0, 0, 1, 0] = 0.5 # p(0,0|1,0)
+    >>> pr_box[1, 1, 1, 0] = 0.5 # p(1,1|1,0)
+    >>> pr_box[0, 1, 1, 1] = 0.5 # p(0,1|1,1)
+    >>> pr_box[1, 0, 1, 1] = 0.5 # p(1,0|1,1)
+    >>> fp_to_fc(pr_box, behaviour=True) # doctest: +NORMALIZE_WHITESPACE
+    array([[ 1.,  0.,  0.],
+           [ 0.,  1.,  1.],
+           [ 0.,  1., -1.]])
+
+    :param v_mat: The probability tensor V[a, b, x, y] in Full Probability notation (oa=2, ob=2).
+    :param behaviour: If True, assume input is a behaviour (default: False, assume functional).
+    :return: The matrix in Full Correlator notation.
+
+    .. note::
+        This function is adapted from the QETLAB MATLAB package function `FP2FC`.
+        For `behaviour=True`, it calculates the *average* marginal correlators `<Ax>` and `<By>`
+        by summing over the other party's inputs and dividing by the number of inputs (ib or ia).
+        The joint correlators `<AxBy>` are calculated directly for each (x, y).
```

Comment:
```suggestion
        This function is adapted from the QETLAB MATLAB package function ``FP2FC``.
        For ``behaviour=True``, it calculates the *average* marginal correlators ``<Ax>`` and `<By>`
        by summing over the other party's inputs and dividing by the number of inputs (ib or ia).
        The joint correlators `<AxBy>` are calculated directly for each (x, y).
```

Reply:
minor nit. 

In the North American English style, `behaviour` is spelled as `behavior`.